### ISSUE TITLE
TFlint doesn't report an error if there are multiple definitions of "required_providers" within a module

### ISSUE THREAD

**Issue by kamlesh-mutha** [terraform-linters/tflint#2227]
### Summary

In a couple of terraform modules, I had inadvertently added "required_providers" block in two different files under the module. 

**main.tf** had the definition as follows:

`terraform {
  required_providers {
    aws = {
      source                = "hashicorp/aws"
      version               = ">= 2.7.0"
    }
  }
}`

**providers.tf**, also under the same module folder, had this definition:

`terraform {
  required_version = "~> 1.5.5"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.2.0"
    }
  }
}`

I've had multiple runs of TFlint on this structure and it never reported any issues with it.

However, **terraform init** immediately reported the following error on the first run using this module:

![Image](https://github.com/user-attachments/assets/ae380021-a1e2-4f70-a1ee-3395b10158c2)

Had tflint pointed that out, it would have saved us some rework.

### Command

docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:rw -v $(pwd):/tmp/lint:rw oxsecurity/megalinter-terraform:v8.4.0

### Terraform Configuration

```terraform
terraform {
  required_providers {
    aws = {
      source                = "hashicorp/aws"
      version               = ">= 2.7.0"
    }
  }
}

terraform {
  required_version = "~> 1.5.5"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.2.0"
    }
  }
}
```

### TFLint Configuration

```HCL
config {
  call_module_type = "local"
  force = false
}

plugin "terraform" {
  enabled = true
  preset  = "recommended"
}

plugin "aws" {
    enabled = true
    version = "0.37.0"
    source  = "github.com/terraform-linters/tflint-ruleset-aws"
}
```

### Output

```Shell
No errors in the tflint output.
```

### TFLint Version

v0.55.0

### Terraform Version

v1.5.7

### Operating System

- [x] Linux
- [x] macOS
- [ ] Windows
